site/ic_data_repo/microsoft_graph_api_client.py

```diff
@@ -0,0 +1,38 @@
+"""Client interface for the Microsoft Graph API."""
+
+from typing import Any, Optional
+
+import requests
+from flask import current_app
+
+
+def _get_app_access_token() -> str:
+    """Get an access token for the application to use the Microsoft Graph API.
+
+    Fetches an access token that is enabled for app-only access i.e. not on behalf of a
+    logged in user.
+    """
+    tenant_id = current_app.config["ICL_OAUTH_WELL_KNOWN_URL"].split("/")[3]
+    response = requests.post(
+        f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token",
+        data={
+            "grant_type": "client_credentials",
+            "client_id": current_app.config["ICL_OAUTH_CLIENT_ID"],
+            "client_secret": current_app.config["ICL_OAUTH_CLIENT_SECRET"],
+            "scope": "https://graph.microsoft.com/.default",
+        },
+    )
```

Comment:
I suppose we could also check the HTTP status... I know things will fail on the next line if the server doesn't return data anyway, but it might make debugging easier if we have a specific check.

```suggestion
    )
    response.raise_for_status()
```